src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MeterListener.cs

```diff
@@ -117,37 +117,39 @@ public void EnableMeasurementEvents(Instrument instrument, object? state = null)
         /// <param name="measurementCallback">The callback which can be used to get measurement recording of numeric type T.</param>
         public void SetMeasurementEventCallback<T>(MeasurementCallback<T>? measurementCallback) where T : struct
         {
-            if (measurementCallback is MeasurementCallback<byte> byteCallback)
+            if (measurementCallback is null)
+                measurementCallback = (instrument, measurement, tags, state) => { /* no-op */};
```

Comment:
nit: could you please add `{` for this if block?

Reply:
I fix it